backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Client.java

```diff
@@ -0,0 +1,219 @@
+    try {
+      stsClient = AWSSecurityTokenServiceClientBuilder.standard().withRegion(awsRegion).build();
+
+      String roleArn = System.getenv(ModelDBConstants.AWS_ROLE_ARN);
+
+      String token =
+          new String(
+              Files.readAllBytes(
+                  Paths.get(
+                      ModelDBUtils.appendOptionalTelepresencePath(
+                          System.getenv(ModelDBConstants.AWS_WEB_IDENTITY_TOKEN_FILE)))));
+
+      // Obtain credentials for the IAM role. Note that you cannot assume the role of
+      // an AWS root account;
+      // Amazon S3 will deny access. You must use credentials for an IAM user or an
+      // IAM role.
+      AssumeRoleWithWebIdentityRequest roleRequest =
+          new AssumeRoleWithWebIdentityRequest()
+              .withDurationSeconds(900)
+              .withRoleArn(roleArn)
+              .withWebIdentityToken(token)
+              .withRoleSessionName(roleSessionName);
+
+      LOGGER.debug("assuming role");
+      // Call STS to assume the role
+      AssumeRoleWithWebIdentityResult roleResponse =
+          stsClient.assumeRoleWithWebIdentity(roleRequest);
+
+      Credentials credentials = roleResponse.getCredentials();
+
+      // Extract the session credentials
+      BasicSessionCredentials awsCredentials =
+          new BasicSessionCredentials(
+              credentials.getAccessKeyId(),
+              credentials.getSecretAccessKey(),
+              credentials.getSessionToken());
+
+      LOGGER.debug("creating new client");
+
+      newS3Client =
+          AmazonS3ClientBuilder.standard()
+              .withCredentials(new AWSStaticCredentialsProvider(awsCredentials))
+              .withRegion(awsRegion)
+              .build();
+
+      newS3Client.doesBucketExistV2(bucketName);
+
+      // Start a thread that will refresh the token. It will just retry for as long as we get an
+      // exception and die right after.
+      LOGGER.debug("scheduling refresh");
+      Thread thread =
+          new Thread(
+              () -> {
+                Long now = System.currentTimeMillis();
+                Long expiration = credentials.getExpiration().getTime();
+                // Wait for half of the duration of the credentials
+                Long waitPeriod = (expiration - now) / 2;
+
+                LOGGER.debug(String.format("sleeping for %d ms", waitPeriod));
+
+                try {
+                  Thread.sleep(waitPeriod);
+                } catch (InterruptedException e) {
+                  // Continue as this just refreshes the session earlier
+                }
+
+                LOGGER.debug("starting refresh");
+
+                // Loop forever until we get to refresh without an exception
+                while (true) {
+                  try {
+                    // Sleep for a second to avoid overwhelming the service
+                    Thread.sleep(1000);
+                    initializetWithTemporaryCredentials(awsRegion);
```

Comment:
```suggestion
                    initializeWithTemporaryCredentials(awsRegion);
```

Reply:
Might need to change in other places too.